File: problems/1544_make_the_string_great.py
class Solution:
    def makeGood(self, s: str) -> str:
        start = 0
        while True:
            start_len = len(s)
            for i in range(start, len(s) - 1):
                if s[i] != s[i + 1] and s[i].lower() == s[i + 1].lower():
                    s = s[0:i] + s[i + 2 :]
                    start = max(0, i - 1)
                    break
            if len(s) == start_len:
                break
        return s

File: problems/1544-make-the-string-great.ts
function makeGood(s: string): string {
  let start = 0
  let startLength = 0

  do {
    startLength = s.length
    for (let i = start; i < s.length - 1; i++) {
      if (s[i].toLowerCase() === s[i + 1].toLowerCase() && s[i] !== s[i + 1]) {
        s = s.slice(0, i) + s.slice(i + 2)
        start = Math.max(0, i - 1)
        break
      }
    }
  } while (startLength !== s.length)

  return s
}

makeGood("abBAcC")
